cmscontrib/loaders/polygon.py

```diff
@@ -145,10 +144,11 @@ def get_task(self, get_statement=True):
         task_cms_conf = None
         if os.path.exists(task_cms_conf_path):
             logger.info("Found additional CMS options for task %s.", name)
-            with open(task_cms_conf_path, 'rb') as f:
-                task_cms_conf = imp.load_module('cms_conf', f,
-                                                task_cms_conf_path,
-                                                ('.py', 'r', imp.PY_SOURCE))
+            import importlib.util
+            spec = importlib.util.spec_from_file_location(
+                'cms_conf', task_cms_conf_path)
+            task_cms_conf = importlib.util.module_from_spec(spec)
+            spec.loader.exec_module(task_cms_conf)
```

Comment:
Have we tried running this on a polygon task by any chance? (TODO for the future: add a test with a polygon task)

Reply:
No, and I don't think it's *too* important as this is a corner-case even in the polygon loader. Anyway, this is documented as equivalent in the python 3.12 docs.

Reply:
Indeed: https://docs.python.org/3.12/library/importlib.html#importing-a-source-file-directly